### PR TITLE
Args: Add ability to specific argType "targets"

### PR DESCRIPTION
**Pull request by tmeasday**
Issue: #16239

## What I did

- Added filter function in store
- Implemented in React
- Added stories to demonstrate

## Still todo

- [ ] Other frameworks
- [ ] Put behind a feature flag somehow.

## How to test

- [x] Is this testable with Jest or Chromatic screenshots?

Yes we have both.

### CODE REVIEW

**Review by shilman — COMMENTED** [review]
Looks great. Do we want to feature flag this?

**Comment by tmeasday** [comment]
Yep! What do you think about the other frameworks? Should I do them too?

**Review by tmeasday — COMMENTED** [review]
@shilman I reworked this, as discussed.

Now:

1. It is a parameter (similar to `passArgsFirst`) rather than a `FEATURE`. This was just more convenient, I am open to turning it back into a feature if you prefer that.
2. We filter `args` by default if the feature is on, also adding `argsByTarget` and `fullArgs` for convenience.

File: lib/store/src/prepareStory.ts [hunk 1]
```diff
@@ -170,12 +171,23 @@ export function prepareStory<TFramework extends AnyFramework>(
...
     }, {} as Args);
-
     const mappedContext = { ...context, args: mappedArgs };
+
+    let finalContext: StoryContext<TFramework> = mappedContext;
+    if (context.parameters.argTypeTarget) {
+      const argsByTarget = groupArgsByTarget({ args: mappedArgs, ...context });
+      finalContext = {
+        ...mappedContext,
+        fullArgs: mappedArgs,
```
Comment (shilman) [inline lib/store/src/prepareStory.ts:181]: `allArgs`?

File: lib/store/src/prepareStory.ts [hunk 2]
```diff
@@ -170,12 +171,23 @@ export function prepareStory<TFramework extends AnyFramework>(
...
+    }
+
     const { passArgsFirst: renderTimePassArgsFirst = true } = context.parameters;
     return renderTimePassArgsFirst
-      ? (render as ArgsStoryFn<TFramework>)(mappedArgs, mappedContext)
-      : (render as LegacyStoryFn<TFramework>)(mappedContext);
+      ? (render as ArgsStoryFn<TFramework>)(finalContext.args, finalContext)
+      : (render as LegacyStoryFn<TFramework>)(finalContext);
   };
   const unboundStoryFn = applyHooks<TFramework>(applyDecorators)(undecoratedStoryFn, decorators);
```
Comment (shilman) [inline lib/store/src/prepareStory.ts:192]: Need to group args here so decorators see the same thing as above

File: lib/store/src/prepareStory.ts [hunk 3]
```diff
@@ -170,12 +171,23 @@ export function prepareStory<TFramework extends AnyFramework>(
       acc[key] = mapping && val in mapping ? mapping[val] : val;
       return acc;
     }, {} as Args);
-
     const mappedContext = { ...context, args: mappedArgs };
+
+    let finalContext: StoryContext<TFramework> = mappedContext;
+    if (context.parameters.argTypeTarget) {
```
Comment (shilman) [inline lib/store/src/prepareStory.ts:177]: `features.argTypeTargetsV7` to denote experimental status?

**Comment by tmeasday** [comment]
@shilman I made the changes you requested.

**Review by shilman — APPROVED** [review]
Love it! 😻

**Pull request merged**